src/mpid/ch4/shm/src/shm_p2p.h

```diff
@@ -12,15 +12,20 @@
 
 MPL_STATIC_INLINE_PREFIX int MPIDI_SHM_mpi_isend(const void *buf, MPI_Aint count,
                                                  MPI_Datatype datatype, int rank, int tag,
-                                                 MPIR_Comm * comm, int context_offset,
+                                                 MPIR_Comm * comm, int attr,
                                                  MPIDI_av_entry_t * addr, MPIR_Request ** request)
 {
     int mpi_errno = MPI_SUCCESS;
 
     MPIR_FUNC_ENTER;
 
-    mpi_errno = MPIDI_IPC_mpi_isend(buf, count, datatype, rank, tag, comm,
-                                    context_offset, addr, request);
+    if (!attr) {
+        /* not collective, not ssend. We are assuming pt2pt context_offset is 0. */
+        mpi_errno = MPIDI_IPC_mpi_isend(buf, count, datatype, rank, tag, comm, attr, addr, request);
+    } else {
+        mpi_errno = MPIDI_POSIX_mpi_isend(buf, count, datatype, rank, tag, comm,
+                                          attr, addr, request);
+    }
```

Comment:
genuinely curious if the non-IPC use cases are intentional, or just an oversight when they were added?

Reply:
The IPC was, I believe, added in a rush as a hack to only treat the non-collective/non-ssend case. When we have a chance we should revisit and clean it up.